pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +99,268 @@ func (builder *Builder) validate() (bool, error) {
+				if condition.Type == conditionType {
+					return condition.Status == isTrue, nil
+				}
+			}
+
+			return false, nil
+		})
+}
+
+// WaitUntilUpdateIsStarted waits until there is a history entry indicating the update start.
+func (builder *Builder) WaitUntilUpdateIsStarted(timeout time.Duration) error {
+	return builder.WaitUntilUpdateHistoryStateTrue("Partial", timeout)
+}
+
+// WaitUntilUpdateIsCompleted waits until there is a history entry indicating the update completed.
+func (builder *Builder) WaitUntilUpdateIsCompleted(timeout time.Duration) error {
+	return builder.WaitUntilUpdateHistoryStateTrue("Completed", timeout)
+}
+
+// WaitUntilUpdateHistoryStateTrue waits until there is a history entry indicating an updateHistoryState.
+func (builder *Builder) WaitUntilUpdateHistoryStateTrue(
+	updateHistoryState v1.UpdateState, timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	if !builder.Exists() {
+		return fmt.Errorf("%s clusterversion not found", builder.Definition.Name)
+	}
+
+	return wait.PollUntilContextTimeout(
+		context.TODO(), time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+			var err error
+			builder.Object, err = builder.Get()
+
+			if err != nil {
+				glog.V(100).Infof("Failed to get the clusterversion with error %s", err)
+
+				return false, nil
+			}
+
+			updateImage := builder.Object.Status.Desired.Image
+
+			for _, updateHistory := range builder.Object.Status.History {
+				if updateHistory.Image == updateImage && updateHistory.State == updateHistoryState {
+					return true, nil
+				}
+			}
+
+			return false, nil
+		})
+}
+
+// GetNextUpdateVersionImage fetches the next recommended or conditional update for the cluster.
+func (builder *Builder) GetNextUpdateVersionImage(stream string, acceptConditionalVersions bool) (string, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Getting the update version image in stream %s for clusterversion %s",
+		stream, builder.Definition.Name)
+
+	if !builder.Exists() {
+		return "", fmt.Errorf("%s clusterversion not found", builder.Definition.Name)
+	}
+
+	if stream == "" {
+		return "", fmt.Errorf("stream can not be empty")
+	}
+
+	currentVersion := builder.Object.Status.Desired.Version
+
+	for _, availableUpdate := range builder.Object.Status.AvailableUpdates {
+		isStreamUpdate, err := builder.isStreamUpdate(currentVersion, availableUpdate.Version, stream)
+		if isStreamUpdate && err == nil {
+			return availableUpdate.Image, nil
+		}
+	}
+
+	if acceptConditionalVersions {
+		for _, conditionalUpdate := range builder.Object.Status.ConditionalUpdates {
+			isStreamUpdate, err := builder.isStreamUpdate(currentVersion, conditionalUpdate.Release.Version, stream)
+			if isStreamUpdate && err == nil {
+				return conditionalUpdate.Release.Image, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("Update version in %s stream not found", stream)
```

Comment:
Please begin error messages with lowercase letters to follow Go best practices
```suggestion
	return "", fmt.Errorf("update version in %s stream not found", stream)
```

Reply:
Fixed, thanks